dsperse/src/run/runner.py

```
@@ -327,36 +283,275 @@ def _generate_run_metadata(self, format: str = "dirs"):
+        slice_idx = tiling.slice_idx
+        cache_input_name = f"tile_{slice_idx}_{tile_idx}_in"
+        cache_output_name = f"tile_{slice_idx}_{tile_idx}_out"
+
+        tile_run_dir = run_dir / slice_id / f"tile_{tile_idx}"
+        tile_run_dir.mkdir(parents=True, exist_ok=True)
+        tile_in = tile_run_dir / "input.json"
+        tile_out = tile_run_dir / "output.json"
+
+        tile_tensor = tensor_cache.get(cache_input_name)
+        if tile_tensor is None:
+            raise ValueError(f"Missing tile input tensor '{cache_input_name}' for slice {slice_id}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)